8.0 document rtf index

NL66278 opened the pull request:
Proper indexing of rtf documents. Can prevent huge performance problems by not indexing embedded images.

NL66278 committed:
[ENH] Add module document_rtf_index.

hbrunn commented on document_rtf_index/README.rst:
```diff
@@ -0,0 +1,37 @@
+Index rtf documents
```
update this to the current template of the oca readme. Main difference is that all links should be https now

hbrunn commented on document_rtf_index/std_index.py:
```diff
@@ -0,0 +1,61 @@
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+import StringIO
+
+from pyth import document
```
protect all non-standard imports

hbrunn commented on document_rtf_index/std_index.py:
```diff
@@ -0,0 +1,61 @@
+        result = _to_unicode(w.getvalue())
+        return result
+
+cntIndex.register(RtfDoc())
```
this means that your code will be used for all databases, not just for the one(s) where the module is installed. Better do this in `_register_hook` of some abstract model, or `ir.attachment` itself

NL66278 replied:
Trouble doing this. Seems _register_hook is only called after tests have been run. Leading to test failure now...

NL66278 replied:
Looking at pre_init hooks now...

NL66278 replied:
No: only runs on new install

NL66278 replied:
In the end just call _register_hook manually from test. No problem as double registration when tests are run will just replace already registered Rtf indexer class by itself.

hbrunn commented on document_rtf_index/tests/test_rtf_index.py:
```diff
@@ -0,0 +1,28 @@
+            'datas_fname': 'test_with_cat_image.rtf',
+            'datas': rtf_file,
+        })
+        self.assertEqual(att1.file_type, 'application/rtf')
```
this is platform dependent. Some distributions return `text/rtf` here, which doesn't seem wrong: https://en.wikipedia.org/wiki/Rich_Text_Format

NL66278 replied:
Not wrong, neither is application/rtf. att1.file_type will be set to the first mime type returned by _getMimeTypes. So in this case not dependent on platform. The test is usefull to make sure it has really been indexed by the rtf indexer, and not by the default one.

hbrunn replied:
thanks for the explanation

hbrunn commented:
an, and in order to get the tests right, pip install your package in `.travis.yml`, and for runbot, we have to ask @gurneyalex nicely to add the package pyth: https://pypi.python.org/pypi/pyth

NL66278 committed:
[FIX] Improvements after review.

- Only load indexer when module is installed;
- Protect non standard imports;
- Standardized README.rst.

NL66278 commented:
I think I added al requested changes now.

gurneyalex commented:
@hbrunn done.

hbrunn commented:
thanks @gurneyalex !
